coderd/rbac/roles_test.go

```diff
@@ -590,6 +590,20 @@ func TestRolePermissions(t *testing.T) {
 				false: {},
 			},
 		},
+		// AnyOrganization tests
+		{
+			Name:     "CreateTemplateAnyOrg",
+			Actions:  []policy.Action{policy.ActionCreate},
+			Resource: rbac.ResourceTemplate.AnyOrganization(),
+			AuthorizeMap: map[bool][]hasAuthSubjects{
+				true: {owner, templateAdmin, orgTemplateAdmin, otherOrgTemplateAdmin, orgAdmin, otherOrgAdmin},
+				false: {
+					userAdmin, memberMe,
+					orgMemberMe, orgAuditor, orgUserAdmin,
+					otherOrgMember, otherOrgAuditor, otherOrgUserAdmin,
+				},
+			},
+		},
```

Comment:
I can add more tests for sure. The test you wrote is missing with `WithOwner`. The test as is, asks the question "Can I create a workspace in any organization that is not owned by me?"

The correct question to see if you can make a workspace belonging to yourself is:

`rbac.ResourceWorkspace.AnyOrganization().WithOwner(currentUser.String())`


The reason the test without the owner works for some roles, is because owners and org admins are able to create workspaces on behalf of other users

Reply:
Ah whoops :+1: